### PR TITLE
Chore: Remove unnecessary files from the docker image

### PR DESCRIPTION
**Pull request by andresmgot**
**What is this feature?**

This PR removes some unnecessary files from the docker image and fixes the docker build when running docker build locally (not sure how it's working on CI tbh).

The specific changes are:

 - Remove `public/plugins/*/pkg` from the final image since Golang files are not required.
 - Avoid building plugins as packages, this prevents from generating a `.yarn` folder that was included in the build.

This reduces the image in ~15MB making it almost the same size that the current `main` (only with a +500KB difference).

**Which issue(s) does this PR fix?**:

Ref https://github.com/grafana/grafana/issues/68973

**Special notes for your reviewer:**

Please check that:
- [X] It works as expected from a user's perspective.
- [X] If this is a pre-GA feature, it is behind a feature toggle.
- [X] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by dsotirakis — APPROVED** [review]
The changes make sense to me, and since this enables you to create the image locally and also you don't break the flow on CI, approved! Thanks @andresmgot!

File: Dockerfile [hunk 1]
```diff
@@ -17,12 +17,12 @@ WORKDIR /tmp/grafana
 COPY package.json yarn.lock .yarnrc.yml ./
 COPY .yarn .yarn
 COPY packages packages
+COPY public public
 COPY plugins-bundled plugins-bundled
 
 RUN yarn install --immutable
 
 COPY tsconfig.json .eslintrc .editorconfig .browserslistrc .prettierrc.js babel.config.json .linguirc ./
-COPY public public
```
Comment (marefr) [inline Dockerfile:25]: What's the reason moving this up?

**Review by marefr — APPROVED** [review]
LGTM but feels like we should make changes somewhere around here to properly resolve this for any build/packaging since that will apply to docker images as well (maybe not local builds though) https://github.com/grafana/grafana/blob/main/pkg/build/packaging/grafana.go#L646. 

Side note: In general, feels like using public/plugins directory is not optimal since we need to remove things from it during build and I would like to explore what possible alternatives might be.

File: Dockerfile (hunk 1 above)
Comment (andresmgot) [reply Dockerfile:25]: `yarn install` fails because the package.json has a reference to `@grafana-plugins/testdata` which has not been copied yet.

**Comment by andresmgot** [comment]
> LGTM but feels like we should make changes somewhere around here to properly resolve this for any build/packaging since that will apply to docker images as well (maybe not local builds though) https://github.com/grafana/grafana/blob/main/pkg/build/packaging/grafana.go#L646.

Do you mean that rather than removing the folder in the Dockerfile we could avoid including it in the grafana package (tar.gz), right? It makes sense so I can add that to this PR (BTW, I'll keep the `rm` in the Dockerfile so it works when building the image from the repository, without a prebuilt package).

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**